Pspcl/Pspcl.Services/StockService.cs

```diff
@@ -506,8 +522,83 @@ public void UpdateStockMaterial(List<List<int>> selectedRowsToDelete)
             }
             _dbcontext.SaveChanges();
             
-        }       
+        }
+
+        public string UploadImageToAzure(IFormFile file)
+        {
+            string fileExtension = Path.GetExtension(file.FileName);
+            using MemoryStream fileUploadStream = new MemoryStream();
+            {
+                file.CopyTo(fileUploadStream);
+                fileUploadStream.Position = 0;
+                BlobContainerClient blobContainerClient = new BlobContainerClient(
+                    _azureOptions.ConnectionString,
+                    _azureOptions.Container);
+
+                var uniqueName = Guid.NewGuid().ToString() + fileExtension;
+
+                try
+                {
+                    //azure package is required for below line
+                    BlobClient blobClient = blobContainerClient.GetBlobClient(uniqueName);
+                    blobClient.Upload(fileUploadStream, new BlobUploadOptions()
+                    {
+                        HttpHeaders = new BlobHttpHeaders
+                        {
+                            ContentType = "image/bitmap"
+                        }
+                    }, cancellationToken: default);
+                
+                }
+                catch (RequestFailedException ex)
+                {
+                    // An exception occurred during the upload
+                    Console.WriteLine("Upload failed. Error message: " + ex.Message);
+                }
+
+                return uniqueName;
+            }
+           
+        }
+        public string DownloadFileFromBlob(string fileName)
+        {
+            CloudStorageAccount account = CloudStorageAccount.Parse("DefaultEndpointsProtocol=https;AccountName=pspclstorage;AccountKey=b4bgeAkYvbYW1W50h03HzZ5B4HWS71fmLgdCwMCTif1gfOmJ8no9eewFjL2oTONwV0kz+NkG0owT+AStlsTshQ==;EndpointSuffix=core.windows.net");
```

Comment:
Configurable settings should be fetched from AppSettings

Reply:
There are multiple hard codes parts of this connection string, which should be picked up from the appsettings.